### PR TITLE
Improve ondemand data collection for npm and maven

### PR DESCRIPTION
**Pull request by pombredanne**
This PR provides various improvements wrt. on-demand data collection, and in particular:
- Increase the priority of ScannableURIs
- https://github.com/nexB/purldb/issues/486
- https://github.com/nexB/purldb/issues/488
- Reintroduce directory fingerprint endpoint
- Fix purl2vcs bug
- Fix https://github.com/nexB/purldb/issues/490
- Enable reindexing of metadata only towards https://github.com/nexB/purldb/issues/494 and https://github.com/nexB/purldb/issues/496

### CODE REVIEW

File: matchcode/api.py
Comment (JonoYang) [inline matchcode/api.py:1]: @pombredanne Should we leave in the directory matching api endpoints for now until we get the scio d2d pipeline to use the matchcode.io matching endpoints?

File: minecode/model_utils.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 )
 
 
-def add_package_to_scan_queue(package, pipelines=DEFAULT_PIPELINES, reindex_uri=False, priority=0):
+def add_package_to_scan_queue(package, pipelines=DEFAULT_PIPELINES, reindex_uri=False, priority=100):
```
Comment (JonoYang) [inline minecode/model_utils.py:47]: @pombredanne When we add something to the scan queue, do we want it to be placed on top of the stack of packages to be scanned, or should that behavior only be done on requests from the `collect` or similar endpoints?

**Review by JonoYang — COMMENTED** [review]
@pombredanne I have some questions about expected behavior

File: matchcode/api.py
Comment (pombredanne) [reply matchcode/api.py:1]: yes

File: minecode/model_utils.py (hunk 1 above)
Comment (pombredanne) [reply minecode/model_utils.py:47]: yes, top of the stack when asking for an exact version IMHO

Comment (JonoYang) [reply minecode/model_utils.py:47]: ack, tracking this in #502

**Pull request merged**